### PR TITLE
Add first to Sequence, deprecate takeFirst

### PR DESCRIPTION
**Pull request by hiltonc**

### CODE REVIEW

File: SwiftificationTests/SequenceTests.swift [hunk 1]
```diff
@@ -32,6 +32,21 @@ func == (lhs: EquatableButNotHashable, rhs: EquatableButNotHashable) -> Bool {
 }
 
 class SequenceTests: XCTestCase {
+    func testFirst1() {
+        let array = [1, 2, 3]
```
Comment (nickmshelley) [inline SwiftificationTests/SequenceTests.swift]: Arrays already have `first`, so I don't think this is testing what you expect.

**Comment by nickmshelley** [comment]
One concern about tests.

Also, out of curiosity, why do you need `first` on `Sequence`?

**Comment by nickmshelley** [comment]
Oh actually I think I remember getting a sequence from some function and having to turn it into an array before being able to call `first`. This would probably be cleaner and more performant.

File: SwiftificationTests/SequenceTests.swift (hunk 1 above)
Comment (hiltonc) [inline SwiftificationTests/SequenceTests.swift]: Oh, right. Thanks!

**Comment by hiltonc** [comment]
I have an infinite Sequence that I want to get the first element from. I can't turn it into an Array.

**Comment by nickmshelley** [comment]
But you can try! 😉

**Comment by hiltonc** [comment]
Just for kicks I gave it a try :). App stalled and memory use goes through the roof. I didn't have patience to wait beyond 20 GB.

**Comment by nickmshelley** [comment]
:+1:

**Pull request merged**